### PR TITLE
Core/Spells: Change to the way the spell effect SPELL_EFFECT_JUMP work. 

### LINKED ISSUE #1141: Big list of commits to port from other projects [$200]
**This is barely formatted, if someone from the team wants to edit this and make it look organized and clean, go on.**

This is a list of commits to port from other projects (mangos, trinity, or any core in fact).

# SUNWELL CORE (Maczuga's)

- [ ] https://github.com/Maczuga/SunwellCore/commit/9fb7429c4310c218a0adbdda45b174ff2db9eb11 Note: with this commit, you will see a buggy hp in princes in ICC so take care
- [ ] https://github.com/Maczuga/SunwellCore/commit/5d9826cac236af7f4601e55f4b10e666fc2f8ebb
- [x] https://github.com/Maczuga/SunwellCore/commit/bacf23abde1251fdc3245688f716403b03923a54
- [ ] https://github.com/Maczuga/SunwellCore/commit/fb94e1d061b58f74ada40aacc545ba9042caccfb
- [ ] https://github.com/Maczuga/SunwellCore/commit/fffefd59fa278b52a1a427479ba0a83024678356
- [x] https://github.com/Maczuga/SunwellCore/commit/5163f56b8242432ce30de204d109cdf59b933a95
- [x] https://github.com/Maczuga/SunwellCore/commit/045a5da4b9ae1909a74a2740151dbc918ffc983c
- [ ] https://github.com/Maczuga/SunwellCore/commit/b00da9c95ccbd5b8edb5d0c4b9a854c062ff7b7e
- [ ] https://github.com/Maczuga/SunwellCore/commit/0f9366b9bfbc73985faa397fe16fe982bb0dc2f2
- [ ] https://github.com/Maczuga/SunwellCore/commit/4cab5e60703d261357e48bd4929d6687cc51701c
- [ ] https://github.com/Maczuga/SunwellCore/commit/5eb4ca77d18de30048541273a68dee4e2af2717b
- [ ] https://github.com/Maczuga/SunwellCore/commit/4e308b87c7a6b09adb27bf00e0d8615302dca4b7
- [ ] https://github.com/Maczuga/SunwellCore/commit/998d1e6448641b6f0fb1da512c567558572010dc
- [x] https://github.com/Maczuga/SunwellCore/commit/0b3db851160db597e5967c101c19bd8fd9a31950
- [ ] https://github.com/Maczuga/SunwellCore/commit/afb89cb7c99fcd90c1fe4bf23f7b57395c8397e2
- [ ] https://github.com/Maczuga/SunwellCore/commit/31a60fdaab1eb4502821fd0dcf6d15bbf11e600e
- [x] https://github.com/Maczuga/SunwellCore/commit/7b1945a5621125111d31d7e4398dc8df94aedeb0
- [ ] https://github.com/Maczuga/SunwellCore/commit/8c11d41e56704f1ad8522cef1b0af2093f342463
- [x] https://github.com/Maczuga/SunwellCore/commit/ae300d8f8fa06b5c7a21ade9e9262e7ed2f1b9a9
- [ ] https://github.com/Maczuga/SunwellCore/commit/065336c79c94411520ac9ee908f59dd7fdbbaf3d
- [ ] https://github.com/Maczuga/SunwellCore/commit/ef25174df07d7fa282e5aaa6e79ee8c0bc0d819e
- [x] https://github.com/Maczuga/SunwellCore/commit/1fdbd9c27bed448b7607a2cacee914269e2bdf52

# TRINITY CORE

Not classified :
- [x] https://github.com/TrinityCore/TrinityCore/commit/a1f2f30c145f6ad9c4baeffeff32618e71ff537c & https://github.com/TrinityCore/TrinityCore/commit/2ff76e828f7c757b5120aba41947a6d63e082e17
- [x] https://github.com/TrinityCore/TrinityCore/commit/d57307f63d8deb51003d61163adccce4e2c1bd47 LOS problem
- [ ] https://github.com/TrinityCore/TrinityCore/commit/59db2eeea0a35028779fd76372ae06cc98c8086f Dynamic respawn NPC / GOB but there are MORE COMMITS fixing this feature on TC
- [ ] https://github.com/TrinityCore/TrinityCore/commit/dae3ae069651ed3f35f975091af59a33181ef0e0 On first login - Styx merged it without any problem
- [x] https://github.com/TrinityCore/TrinityCore/commit/7dfd472f8dff6ce067572e2887c2beb56d9ba9de Proper client timeout detection (fall anticheat)
- [ ] https://github.com/TrinityCore/TrinityCore/commit/508c9d2fc1b20dc2cb40df533e823e1dfe2becc3  Better addon communication
- [ ] https://github.com/TrinityCore/TrinityCore/commit/ce947814c6fa2c32c0908f1e7088ce971f74d365 https://github.com/azerothcore/azerothcore-wotlk/issues/1506

IMPORTANT:

- [ ] https://github.com/TrinityCore/TrinityCore/pull/18771 SEEMS RATHER IMPORTANT (prob hard)
- [ ] https://github.com/TrinityCore/TrinityCore/pull/19914 
- [ ] https://github.com/TrinityCore/TrinityCore/pull/19767/files
- [ ] https://github.com/TrinityCore/TrinityCore/pull/21282/ Core/Creature: implement automatic distance correction for melee combat
- [ ] https://github.com/TrinityCore/TrinityCore/pull/18128 Core/Movement: Using player position extrapolation to smooth range checks
- [ ] https://github.com/TrinityCore/TrinityCore/pull/21193 Core/Movement: Correct the allowed distance to target before a repositioning is necessary.
- [x] https://github.com/TrinityCore/TrinityCore/commit/469f2d3ade168de0b5be18112008c337067a8e03 CMake: Use inherited dependencies in database, shared and game
- [x] https://github.com/TrinityCore/TrinityCore/commit/2a3546ca36468b81d673bdb27315006144af3165
Core/Skills: Use SkillLineAbility.dbc to determine player initial spells  - skill assignment done in a new table `playercreateinfo_skills`
- [ ] https://github.com/TrinityCore/TrinityCore/pull/23864  NPC movement: implement smooth waypoint transition 

MINOR - EASY TO ADD :

- [ ] https://github.com/TrinityCore/TrinityCore/pull/20305/commits/6b226cf735e23bdb40c2dcaba644e0ebbb620cb2
- [x] https://github.com/TrinityCore/TrinityCore/pull/19438 NEED TO SEE IF IT BREAKS OTHER SPELLS (added here https://github.com/azerothcore/azerothcore-wotlk/pull/1696 )
- [x] https://github.com/TrinityCore/TrinityCore/pull/19055/files
- [x] https://github.com/TrinityCore/TrinityCore/commit/62f7debffdb0a45cec0744ddb29460214febc9b8
- [x] https://github.com/TrinityCore/TrinityCore/commit/1a6d0b8d951534754c5a82a7a861f415c1b2f2ca
- [x] https://github.com/TrinityCore/TrinityCore/commit/4d654f548b29bf61552d9f90da8254ecea91b63e
- [x] https://github.com/TrinityCore/TrinityCore/commit/015c6b56e965d3c7f80d5109915237482c631b03
- [x] https://github.com/TrinityCore/TrinityCore/pull/20202/commits/f455d9db1f9e461cb0090ddd4d95f40a2edac11d For neutral guard AI? Added here but didn't seem very useful https://github.com/azerothcore/azerothcore-wotlk/pull/2643
- [ ] https://github.com/TrinityCore/TrinityCore/pull/19352/files Add missing parameter TARGET to smart_scripts 
- [ ] https://github.com/TrinityCore/TrinityCore/commit/132538db1dfb3c495294eeb5acfead59dea02b5c Better grouphandling - Modify group invite behavior while creating group to be blizzlike

# DALARAN CORE

- [ ] https://github.com/Falstring/Dalaran-Core/blob/3605a709de0b6b73c05dcb0bd2eb772d7b0bcf86/src/server/scripts/World/zeppelins.cpp  Zeppelins - Related to this PR https://github.com/azerothcore/azerothcore-wotlk/pull/907 () removed gossip menu options 1969 and 1971  - need to put them back with missing zeppelin master gossip menus)

# OTHERS

- [ ] CLUSTERING

Got idea from Fall1n who has done it on SunwellCore in his private core
https://www.reddit.com/r/wowservers/comments/8u1s8i/spell_attro_cu_fake_delay_does_it_work_only_on/e1c2ol6/
https://bitbucket.org/lukaasm/hellground-wowemu/commits/d09c115242ee9f57da7eed11e8c764f7faeb2f70

- [ ] CROSS REALMS

On a firestorms leak (legion/mop) like [DynastyCore](https://github.com/TheGhostGroup/DynastyCore)

- [ ] Funny command for GM to mount any creature

https://github.com/vmangos/core/commit/b53a0f10b77a8a46d5621c4ca8ee4614553cc516 it's already possible ingame but the command is rather complex, i don't remember it exactly

<bountysource-plugin>

---
There is a **[$200 open bounty](https://www.bountysource.com/issues/67930495-big-list-of-commits-to-port-from-other-projects?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:
-  changed the wat how get targets for SPELL_EFFECT_JUMP
##### Issue adressed: close #1141

### CODE REVIEW

**Comment by comix1988** [comment]
@Viste please, can you give some examples of creatures to check?

File: _config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+theme: jekyll-theme-slate
```
Comment (FrancescoBorzi) [inline _config.yml]: ?

Comment (Viste) [reply _config.yml]: It's an accident :S

**Comment by GitVerge** [comment]
@comix1988 this works for players or creatures. For instance, use death grip and check how they are pulled (speed, height, etc). Also in lich king, after killing all the raid when that npc is released from the ice and jumps with the sword.

**Comment by sucofog** [comment]
@GitVerge Would this make any difference or improvement?

**Comment by GitVerge** [comment]
It could make a difference when the target is not within LoS and when position Z difference between target and caster is bigger than 3

**Comment by comix1988** [comment]
I tested 6 spells:
3 spells with SPELL_EFFECT_JUMP:
- HEROIC LEAP used by "defender of the light" spell ID 53625
- JUMP ATTACK used by "gundrak raptor" spell ID 54487
- BELLY FLOP used by "king ping" spell ID 61115

3 spells with SPELL_EFFECT_JUMP_DEST:
- STOMP used by "bonegrinder" spell ID 56737
- SPIRIT WOLF LEAP used by Shaman's "spirit wolfs" spell ID 58867
- POUNCE used by "loque'nahak" Spell ID 61184

I have tested the ranges of use and their mechanism and everything works correctly. Good job @Viste
p.s.: spell death grip is not affected by jump effect

**Comment by Stoabrogga** [comment]
Thanks @Viste for providing this PR and @comix1988 for testing it!

**Comment by GitVerge** [comment]
@comix1988 looks like death grip is affected by jump dest instead of only jump, but was not changed in this commit so is fine ^^

**Pull request merged**